python/PythonMessage.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of libArcus
+ *
+ * Copyright (C) 2016 Ultimaker b.v. <a.hiemstra@ultimaker.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ARCUS_MESSAGE_PTR_H
+#define ARCUS_MESSAGE_PTR_H
+
+#include <Python.h>
+#include "Types.h"
+
+namespace google
+{
+    namespace protobuf
+    {
+        class Descriptor;
+        class Reflection;
+    }
+}
+
+namespace Arcus
+{
+    /**
+     * A simple wrapper around a Protobuf message so it can be used from Python.
+     *
+     * This class wraps a Protobuf message and makes it possible to get and set
+     * values from the message. Message properties are exposed as Python protoperties
+     * so can be set using things like `message.data = b"something"` from Python.
+     *
+     * Repeated messages are supported, using addRepeatedMessage, repeatedMessageCount
+     * and getRepeatedMessage. A repeated message is returned as a PythonMessage object
+     * so exposes the same API as the top level message.
+     */
+    class PythonMessage
+    {
+    public:
+        PythonMessage(google::protobuf::Message* message);
+        PythonMessage(const MessagePtr& message);
+        virtual ~PythonMessage();
+
+        /**
+         * Get the message type name of this message.
+         */
+        std::string getTypeName() const;
+
+        /**
+         * Python property interface.
+         */
+        bool __hasattr__(const std::string& field_name) const;
+        PyObject* __getattr__(const std::string& field_name) const;
+        void __setattr__(const std::string& name, PyObject* value);
+
+        /**
+         * Add an instance of a Repeated Message to a specific field.
+         *
+         * \param field_name The name of the field to add a message to.
+         *
+         * \return An instance of PythonMessage wrapping the new Message in the field.
```

Comment:
That it doesn't return an instance, it returns a pointer to it. It's a bit nitpicking.